jerry-core/lit/lit-char-helpers.cpp

```diff
@@ -313,6 +313,51 @@ lit_char_hex_to_int (ecma_char_t c) /**< code unit, corresponding to
 } /* lit_char_hex_to_int */
 
 /**
+ * Parse the next number_of_characters hexadecimal character,
+ * and construct a code point from them. The buffer must
+ * be zero terminated.
+ *
+ * @return the decoded character code
+ *         It returns with ECMA_BUILTIN_HEX_TO_CODE_POINT_ERROR if a parse error is occured.
+ */
+lit_code_point_t
+lit_read_code_point_from_hex (const lit_utf8_byte_t *buf_p, /**< buffer with characters */
+                              lit_utf8_size_t number_of_characters) /**< number of characters to be read */
+{
+  uint32_t character_code = 0;
+
+  JERRY_ASSERT (number_of_characters >= 2 && number_of_characters <= 4);
+
+  for (lit_utf8_size_t i = 0; i < number_of_characters; i++)
+  {
+    character_code <<= 4;
+
+    if (*buf_p >= LIT_CHAR_ASCII_DIGITS_BEGIN
+        && *buf_p <= LIT_CHAR_ASCII_DIGITS_END)
+    {
+      character_code |= (uint32_t) (*buf_p - LIT_CHAR_ASCII_DIGITS_BEGIN);
+    }
+    else if (*buf_p >= LIT_CHAR_ASCII_LOWERCASE_LETTERS_HEX_BEGIN
+             && *buf_p <= LIT_CHAR_ASCII_LOWERCASE_LETTERS_HEX_END)
+    {
+      character_code |= (uint32_t) (*buf_p - (LIT_CHAR_ASCII_LOWERCASE_LETTERS_HEX_BEGIN - 10));
+    }
+    else if (*buf_p >= LIT_CHAR_ASCII_UPPERCASE_LETTERS_HEX_BEGIN
+             && *buf_p <= LIT_CHAR_ASCII_UPPERCASE_LETTERS_HEX_END)
+    {
+      character_code |= (uint32_t) (*buf_p - (LIT_CHAR_ASCII_UPPERCASE_LETTERS_HEX_BEGIN - 10));
+    }
+    else
+    {
+      return ECMA_BUILTIN_HEX_TO_CODE_POINT_ERROR;
```

Comment:
Could you, please, add `out_code_point_p` argument for returning code point, and change return value type to `true` / `false` for indicating whether conversion was performed successfully? In the case, `ECMA_BUILTIN_HEX_TO_CODE_POINT_ERROR` would be unnecessary.